python/lsst/sims/photUtils/Sed.py

```diff
@@ -636,15 +654,32 @@ def multiplySED(self, other_sed, wavelen_step=PhotometricDefaults.wavelenstep):
     ## routines related to magnitudes and fluxes
 
     def calcADU(self, bandpass, wavelen=None, fnu=None,
-                expTime=PhotometricDefaults.exptime, effarea=PhotometricDefaults.effarea,
-                gain=PhotometricDefaults.gain):
+                photParams=None):
         """
         Calculate the number of adu from camera, using sb and fnu.
 
         Given wavelen/fnu arrays or use self. Self or passed wavelen/fnu arrays will be unchanged.
         Calculating the AB mag requires the wavelen/fnu pair to be on the same grid as bandpass;
          (temporary values of these are used).
+
+        @param [in] bandpass is an instantiation of the Bandpass class
+
+        @param [in] wavelen (optional) is the wavelength grid in nm
+
+        @param [in] fnu (optional) is the flux in Janskys
+
+        If wavelen and fnu are not specified, this will just use self.wavelen and
+        self.fnu
+
+        @param [in] photParams is an instantiation of the
+        PhotometricParameters class that carries details about the
+        photometric response of the telescope.
         """
+
+        if photParams is None:
```

Comment:
I don't think this is how I would handle photParams. If it's not optional, don't make it an optional kwarg, make it an argument. 
Or you could make it optional, and have the class itself import the "basic default" photometricParameters class in the init? (given that it's not optional in Bandpass however, I think making it not optional here and an argument to the method, is the right approach).